tests/nominal/test_OneHotEncodingTransformer.py

```diff
@@ -142,97 +59,13 @@ def test_fields_with_over_100_levels_error(self):
         ):
             x.fit(df)
 
-    def test_fit_returns_self(self):
-        """Test fit returns self?."""
-        df = d.create_df_1()
-
-        x = OneHotEncodingTransformer(columns="b")
-
-        x_fitted = x.fit(df)
-
-        assert (
-            x_fitted is x
-        ), "Returned value from OneHotEncodingTransformer.fit not as expected."
-
-    def test_fit_not_changing_data(self):
-        """Test fit does not change X."""
-        df = d.create_df_1()
-
-        x = OneHotEncodingTransformer(columns="b")
-
-        x.fit(df)
-
-        ta.equality.assert_equal_dispatch(
-            expected=d.create_df_1(),
-            actual=df,
-            msg="Check X not changing during fit",
-        )
-
-
-class TestTransform:
-    """Tests for OneHotEncodingTransformer.transform()."""
-
-    def expected_df_1():
-        """Expected output for test_expected_output."""
-        df = pd.DataFrame(
-            {
-                "a": [4, 2, 2, 1, 3],
-                "b": ["x", "z", "y", "x", "x"],
-                "c": ["c", "a", "a", "c", "b"],
-            },
-        )
-
-        df["c"] = df["c"].astype("category")
-
-        df["b_x"] = [1.0, 0.0, 0.0, 1.0, 1.0]
-        df["b_y"] = [0.0, 0.0, 1.0, 0.0, 0.0]
-        df["b_z"] = [0.0, 1.0, 0.0, 0.0, 0.0]
-
-        return df
-
-    def expected_df_2():
-        """Expected output for test_unseen_categories_encoded_as_all_zeroes."""
-        df = pd.DataFrame(
-            {
-                "a": [1, 5, 2, 3, 3],
-                "b": ["w", "w", "z", "y", "x"],
-                "c": ["a", "a", "c", "b", "a"],
-            },
-            index=[10, 15, 200, 251, 59],
-        )
-
-        df["c"] = df["c"].astype("category")
-
-        df["a_1"] = [1.0, 0.0, 0.0, 0.0, 0.0]
-        df["a_2"] = [0.0, 0.0, 1.0, 0.0, 0.0]
-        df["a_3"] = [0.0, 0.0, 0.0, 1.0, 1.0]
-        df["a_4"] = [0.0, 0.0, 0.0, 0.0, 0.0]
-        df["b_x"] = [0.0, 0.0, 0.0, 0.0, 1.0]
-        df["b_y"] = [0.0, 0.0, 0.0, 1.0, 0.0]
-        df["b_z"] = [0.0, 0.0, 1.0, 0.0, 0.0]
-        df["c_a"] = [1.0, 1.0, 0.0, 0.0, 1.0]
-        df["c_b"] = [0.0, 0.0, 0.0, 1.0, 0.0]
-        df["c_c"] = [0.0, 0.0, 1.0, 0.0, 0.0]
-
-        return df
-
-    def test_columns_check_call(self, mocker):
-        """Test the first call to BaseTransformer columns_check."""
-        df = d.create_df_1()
 
-        x = OneHotEncodingTransformer(columns="b")
+class TestTransform(DropOriginalTransformMixinTests, GenericTransformTests):
```

Comment:
Think this wants to inherit from test_BaseNominalTransformer.GenericBaseNominalTransformerTests (which in other parallel PRS #259 #261 I have renamed to GenericNominalTransformTests because the name is currently a bit confusing and not following the pattern set elsewhere)

Reply:
Hey David - might need to discuss this further. On the 'test_not_fitted_error_raised' which is inherited from 'GenericBaseNominalTransformerTests' the test fails for OHE as trying to call transform without fitting throws the following error for OHE 'AttributeError: 'OneHotEncodingTransformer' object has no attribute 'categories_''' - I came across this with some other tests but added in a fit method to resolve. 

Also the 'test_exception_raised' passes due to the behaviour of OHE, which I don't think is unexpected (although might not have thought it through properly). I know your off today so we can catch up next week!

Reply:
Options here are to either fix the code to generate the same error as the test is expecting  or to overwrite the test (i.e. create a test in the child test class with the same name that just contains` pass`, which will then supersede the test from the parent).

For test_not_fitted_error_raised I think we should add in a check that the transformer is fitted which should be caught before the call to sklearn.

For the other one, I suggest we overwrite the test.  Note this test has been given a more descriptive name in #259 `test_non_mappable_rows_exception_raised`

Reply:
Thanks David - have addressed these comments [here](https://github.com/lvgig/tubular/pull/267/commits/d2ffa719e974345a86a86ab7b9f3893f5bc1bb26).

In terms of the 'test_exception_raised' (which I know has been renamed in another PR so will need to be resolved when merging) I have overwritten the test as suggested.

After our conversation in relation to 'test_not_fitted_error_raised ' I have used the check_is_fitted wrapper we have created for the sklearn.utils check_is_fitted function to check the attribute '_categories' exists before the .transform method is called. '_categories' is created by the Sklearn base encoder so will only be there if our call to the OHE transformer is successfully fit.

As per our conversation, this **has introduced a dependency on the sklearn OHE implementation.** Think we are both agreed that this is something we would rather catch so that if OHE is changed, we get notification rather than a change of behaviour not highlighted. Added this as a note to the changelog also.

Reply:
Renamed 'test_exception_raised' [here](https://github.com/lvgig/tubular/pull/267/commits/1e1e6e41223321c59e0ac5f1d21e239e7db91bc9) - didn't realise new naming convention was already in main!

Reply:
Thanks Sam.